src/index.cpp

```diff
@@ -2266,7 +2241,7 @@ std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::search_with_filters(const
     // REFACTOR
     // T *aligned_query = scratch->aligned_query();
     // memcpy(aligned_query, query, _dim * sizeof(T));
-    _distance->preprocess_query(query, _data_store->get_dims(), scratch->aligned_query());
+    _data_store->get_dist_fn()->preprocess_query(query, _data_store->get_dims(), scratch->aligned_query());
```

Comment:
I think the _data_store() has a preprocess_query() so you can invoke it directly.

Reply:
same comment as below. It is not exposed currently through data_store. Keeping this open in case we want to move it.